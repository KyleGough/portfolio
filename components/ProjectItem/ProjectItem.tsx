import { Section } from '@components/Section';
import { getDateRange } from '@utilities/date';
import { ProjectDate } from '@utilities/types';
import Image, { StaticImageData } from 'next/image';
import Link from 'next/link';
import React from 'react';

interface ProjectItemProps {
  title: string;
  date: ProjectDate;
  description: string;
  link: string;
  image: StaticImageData;
  alt: string;
  logo: {
    src: string;
    alt: string;
  };
}

export const ProjectItem: React.FC<ProjectItemProps> = ({
  title,
  date,
  description,
  link,
  image,
  alt,
  logo,
}) => (
  <Section>
    <div className="grid grid-cols-12 gap-8 lg:gap-16">
      <div className="col-span-12 lg:col-span-7">
        <h2 className="mb-4 project-header">{title}</h2>
        <p className="my-4 text-link-hover">{getDateRange(date)}</p>
        <p className="my-4">{description}</p>
      </div>
      <div className="col-span-12 lg:col-span-5 text-center -mb-8">
        <Link href={link}>
          <div className="block group relative">
            <Image
              className="border-2 border-white rounded-lg shadow"
              src={image.src}
              alt={alt}
              width={image.width}
              height={image.height}
              placeholder="blur"
              blurDataURL={image.blurDataURL}
            />
            <div className="absolute top-2 right-2 w-8 h-8 drop-shadow">
              <Image src={logo.src} alt={logo.alt} width={30} height={30} />
            </div>
            <button
              tabIndex={-1}
              className="relative -top-[1.6rem] px-12 py-3 bg-background text-link group-hover:text-link-hover group-focus:text-link-hover rounded-sm shadow border-light whitespace-nowrap"
            >
              View Project
            </button>
          </div>
        </Link>
      </div>
    </div>
  </Section>
);
