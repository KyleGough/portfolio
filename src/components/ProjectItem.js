import React from 'react';
import { Link } from 'react-router-dom';
import Divider from './Divider';
import Section from './Section';

export default function ProjectItem(props) {
    return (
        <>
        <Divider />
        <Section>
            <div className='grid grid-cols-12 gap-8'>
                <div className='col-span-12 lg:col-span-7'>
                    <h2 className='mb-4 project-header'>{props.title}</h2>
                    <p className='my-4 text-link-hover'>{props.date}</p>
                    <p className='my-4'>{props.description}</p>
                </div>
                <div className='col-span-12 lg:col-span-5 text-center -mb-8'>
                    <Link className='project-link' to={props.link}>
                        <img className='border-2 border-link rounded-lg' src={props.src} alt={props.alt} />
                        <button className='relative -top-7 rounded-full border-2 border-disabled bg-gradient-to-r from-link to-sky-600 py-3 px-12 text-white'>View Project</button>
                    </Link>
                </div>
            </div>
        </Section>
        </>
    );
}