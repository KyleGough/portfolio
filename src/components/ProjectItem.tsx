import { Link } from 'react-router-dom';
import Section from './Section';

type ProjectItemProps = {
    title: string,
    date: string,
    description: string,
    link: string,
    src: string,
    alt: string
};

export default function ProjectItem(props: ProjectItemProps) {
    return (
        <Section>
            <div className='grid grid-cols-12 gap-8 lg:gap-16'>
                <div className='col-span-12 lg:col-span-7'>
                    <h2 className='mb-4 project-header'>{props.title}</h2>
                    <p className='my-4 text-link-hover'>{props.date}</p>
                    <p className='my-4'>{props.description}</p>
                </div>
                <div className='col-span-12 lg:col-span-5 text-center -mb-8'>
                    <Link className='group relative' to={props.link}>
                        <img loading='lazy' className='border-2 border-white rounded-lg shadow' src={props.src} alt={props.alt} />   
                        <button tabIndex={-1} className='relative -top-[1.6rem] px-12 py-3 bg-background text-link group-hover:text-link-hover group-focus:text-link-hover rounded-sm shadow border-light whitespace-nowrap'>View Project</button>
                    </Link>
                </div>
            </div>
        </Section>
    );
}
