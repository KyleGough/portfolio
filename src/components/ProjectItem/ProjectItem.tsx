import React from 'react';
import { Link } from 'react-router-dom';

import { IProjectDate } from '../../data';
import { getDateRange } from '../../utilities/date';
import { Section } from '../Section';

interface ProjectItemProps {
  title: string;
  date: IProjectDate;
  description: string;
  link: string;
  src: string;
  alt: string;
  logo: string;
}

export const ProjectItem: React.FC<ProjectItemProps> = ({
  title,
  date,
  description,
  link,
  src,
  alt,
  logo,
}) => (
  <Section>
    <div className="grid grid-cols-12 gap-8 lg:gap-16">
      <div className="col-span-12 lg:col-span-7">
        <h2 className="mb-4 project-header">{title}</h2>
        <p className="my-4 text-link-hover">{getDateRange(date)}</p>
        <p className="my-4">{description}</p>
      </div>
      <div className="col-span-12 lg:col-span-5 text-center -mb-8">
        <Link className="block group relative" to={link}>
          <img
            loading="lazy"
            className="border-2 border-white rounded-lg shadow"
            src={src}
            alt={alt}
          />
          <img className="absolute top-2 right-2 w-8 h-8 drop-shadow" src={logo} />
          <button
            tabIndex={-1}
            className="relative -top-[1.6rem] px-12 py-3 bg-background text-link group-hover:text-link-hover group-focus:text-link-hover rounded-sm shadow border-light whitespace-nowrap"
          >
            View Project
          </button>
        </Link>
      </div>
    </div>
  </Section>
);
