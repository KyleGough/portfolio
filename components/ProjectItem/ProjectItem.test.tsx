import { render, screen } from '@testing-library/react';
import React from 'react';

import { ProjectItem } from './ProjectItem';

describe('ProjectItem component', () => {
  it('renders', async () => {
    const { container } = render(
      <ProjectItem
        title="New Project"
        date={{
          start: { month: 1, year: 2021 },
          end: { month: 6, year: 2022 },
        }}
        description="Sample Description"
        link="/#project"
        image={{
          src: '/#image',
          height: 100,
          width: 100,
          blurDataURL: '/#blur',
        }}
        alt="Test Alt Text"
        logo={{ src: '/#logo', alt: 'Logo' }}
      />
    );
    expect(screen.getByText('January 2021 - June 2022')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
