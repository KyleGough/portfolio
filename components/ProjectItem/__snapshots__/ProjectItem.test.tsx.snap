// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="grid grid-cols-12 gap-8 lg:gap-16"
    >
      <div
        class="col-span-12 lg:col-span-7"
      >
        <h2
          class="mb-4 project-header"
        >
          New Project
        </h2>
        <p
          class="my-4 text-link-hover"
        >
          January 2021 - June 2022
        </p>
        <p
          class="my-4"
        >
          Sample Description
        </p>
      </div>
      <div
        class="col-span-12 lg:col-span-5 text-center -mb-8"
      >
        <a
          href="/#project"
        >
          <div
            class="block group relative"
          >
            <img
              alt="Test Alt Text"
              class="border-2 border-white rounded-lg shadow"
              data-nimg="1"
              decoding="async"
              height="100"
              loading="lazy"
              src="/_next/image?url=%2F%23image&w=256&q=75"
              srcset="/_next/image?url=%2F%23image&w=128&q=75 1x, /_next/image?url=%2F%23image&w=256&q=75 2x"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="100"
            />
            <div
              class="absolute top-2 right-2 w-8 h-8 drop-shadow"
            >
              <img
                alt="Logo"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=%2F%23logo&w=64&q=75"
                srcset="/_next/image?url=%2F%23logo&w=32&q=75 1x, /_next/image?url=%2F%23logo&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
            </div>
            <button
              class="relative -top-[1.6rem] px-12 py-3 bg-background text-link group-hover:text-link-hover group-focus:text-link-hover rounded-sm shadow border-light whitespace-nowrap"
              tabindex="-1"
            >
              View Project
            </button>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
